ui/src/api/api-fetch.ts

```diff
@@ -16,15 +19,40 @@ function getCsrfCookie(): string | undefined {
 }
 
 const TOKEN_EXPIRED = 'error.unauthorized.token-expired';
+const INVALID_VERSION = 'error.bad-request.invalid-version';
+
+const GEOVIITE_UI_VERSION_HEADER_KEY = 'x-geoviite-ui-version';
+
+let geoviiteUiVersion = '';
+const unsubscribeFromStateUpdates = appStore.subscribe(() => {
+    geoviiteUiVersion = appStore.getState().common.version ?? '';
+
+    // Only the version state update after the backend environment info has been fetched for the first time
+    // during loading is meaningful. After the version has been fetched, the ui bundle and environment info won't
+    // be updated without a page refresh.
+    if (geoviiteUiVersion.length > 0) {
+        unsubscribeFromStateUpdates();
+    }
+});
```

Comment:
Sarjassamme ei mielellään näin jos jokin muu rakenne olisi toimivampi... Ehdotuksia?

Reply:
Joo laitanpas sen suosiolla vaan hakemaan sen reduksista joka kerta. Tuntuu vaan himpun typerältä hakea redux-storesta tavaraa melko usein (joka pyynnöllä), joka ei muutu ui:n alustuksen jälkeen koskaan, mutta eipä siitä merkittävää hitauttakaan näyttäisi aiheutuvan.